### ISSUE TITLE
Issues with eagles

### ISSUE THREAD

**Issue by Emicon97** [cigumo/krvi#455]
- [x] Goblirangs miss (at least visually, but I'm almost certain they don't even hit them) when they're attacking a player's unit.
- [ ] Sometimes they just become ignored by towers and Tesla Coils, so they have to be killed by ranged units.
- [ ] It looks really weird when crows attack them, since the crow flies much higher than them.

**Comment by Emicon97** [comment]
1.
https://github.com/ironhidegames/issues-kr4/assets/94487044/c6256f3c-48f7-4583-b40e-b581e25f13a5

3.
![20240222234447_1](https://github.com/ironhidegames/issues-kr4/assets/94487044/584e465c-d775-4ebd-bd06-9c3c1fc1e1ef)

The second issue I listed is a bit harder to reproduce, and I'm not sure how to trigger it, but believe me, it happens very often. And I'm almost sure this happens after they attacked the player's units.

**Comment by Mmlh1** [comment]
For 1. see #419

**Comment by Mmlh1** [comment]
https://github.com/user-attachments/assets/45b9d46d-ead9-490a-b853-1e2ad5f06275

I don't know what the hell is going on, but clearly, their targeting position and hit position are different, and when they somehow get hit by the Frozen Breath, they ignore everything for a few seconds after being unfrozen and then do weird stuff? Please look into this, they are so janky and not fun to play against.

**Comment by Mmlh1** [comment]
Changes paths completely unnecessarily:
https://github.com/user-attachments/assets/69c75067-2ed3-4e7f-8c8f-1bd0c5b90140

Prioritizes the Falconer even though the Eagle is clearly closer to the exit:
https://github.com/user-attachments/assets/c10766fe-a495-4926-8f26-f8d6f30bdb2f

**Comment by pablorealini** [comment]
> Kingdom.Rush.Vengeance.2025-03-30.18-58-58.online-video-cutter.com.mp4 
> 
> I don't know what the hell is going on, but clearly, their targeting position and hit position are different, and when they somehow get hit by the Frozen Breath, they ignore everything for a few seconds after being unfrozen and then do weird stuff? Please look into this, they are so janky and not fun to play against.

I already "fixed" this issue... or at least improved it a little bit.
But should Jig's Ice Ball attack flying units? I don't think so, since the ball hits the ground. What do you think?

**Comment by Mmlh1** [comment]
I don't really like taking away its flying targeting. Would it be possible to target the unit rather than the floor instead? I think that would make sense, and I believe that's one line in the .plist file

**Comment by pablorealini** [comment]
Yes, I already did something like that....sadly I don't have an air explosion for that projectile...but it works.
